peer.go

```diff
@@ -869,8 +869,75 @@ func (p *peer) handleLocalClose(req *closeLinkReq) {
 		}
 		return
 	}
+}
+
+// handleShutdownResponse is called when a responder in a cooperative channel
+// close workflow receives a Shutdown message. This is the second step in the
+// cooperative close workflow. This function generates a close transaction with
+// a proposed fee amount and sends the signed transaction to the initiator.
+func (p *peer) handleShutdownResponse(msg *lnwire.Shutdown) {
+	p.activeChanMtx.RLock()
+	channel, ok := p.activeChannels[msg.ChannelID]
+	p.activeChanMtx.RUnlock()
+	if !ok {
+		peerLog.Errorf("unable to close channel, ChannelPoint(%v) is "+
+			"unknown", msg.ChannelID)
+		return
+	}
+	err := p.sendShutdown(channel)
+	if err != nil {
+		peerLog.Errorf("error while sending shutdown message: %v", err)
+		return
+	}
+
+	// TODO(bvu): add logic to wait for HTLCs to close here.
+	err = p.sendClosingSigned(channel)
+	if err != nil {
+		peerLog.Errorf("error generating/sending ClosingSigned: %v", err)
+		return
+	}
+	return
+}
+
+// handleInitClosingSigned is called when the initiator in a cooperative
+// channel close workflow receives a ClosingSigned message from the responder.
+// This function sends back a corresponding ClosingSigned message, then
+// broadcasts the channel close transaction. It also performs channel cleanup
+// and reports status back to the caller. This is the initiator's final step in
+// the channel close workflow.
+func (p *peer) handleInitClosingSigned(req *closeLinkReq, msg *lnwire.ClosingSigned) {
+	p.activeChanMtx.RLock()
+	channel := p.activeChannels[*req.chanPoint]
+	p.activeChanMtx.RUnlock()
+
+	err := p.sendClosingSigned(channel)
```

Comment:
Atm work is duplicated a bit since on each channel closing workflow, each side will _always_ generate their signature twice. In order to fix this, we can expand the newly added local scope map to `channelManager` to also hold one half of the closing signature spending on context and execution order. Next, we can modify the cooperative closure methods on the channel to have the `CompleteCoopertiveClose` method accept _both_ signatures.